apps/bbsync/tests/test_query.py

```diff
@@ -201,18 +201,58 @@ def test_generate_summary_removed_cve(self):
             (WorkflowModel.WorkflowState.PRE_SECONDARY_ASSESSMENT, "vulnerability"),
             (WorkflowModel.WorkflowState.SECONDARY_ASSESSMENT, "vulnerability"),
             (WorkflowModel.WorkflowState.DONE, "vulnerability"),
-            (WorkflowModel.WorkflowState.REJECTED, "vulnerability"),
+            (WorkflowModel.WorkflowState.REJECTED, "vulnerability-draft"),
         ],
     )
-    def test_generate_component(self, workflow_state, result):
+    @pytest.mark.parametrize("has_meta_attr", [True, False])
+    def test_generate_component_before_bz_sync_or_with_draft_component(
+        self, workflow_state, result, has_meta_attr
+    ):
         """
-        test generating of component
+        Test that component is set to "vulnerability-draft" in NEW and REJECTED
+        workflow states and to "vulnerability" in other states when the flaw
+        hasn't been synced to bugzilla yet or when its component was previously
+        set to "vulnerability-draft".
         """
-        flaw = FlawFactory(workflow_state=workflow_state)
+        if has_meta_attr:
+            flaw = FlawFactory(
+                workflow_state=workflow_state,
+                meta_attr={"bz_id": "123", "bz_component": "vulnerability-draft"},
+            )
+        else:
+            flaw = FlawFactory(workflow_state=workflow_state)
```

Comment:
nitpick: better to test with defined `meta_attr` because `meta_attr` are set via factory so we should explicitly state what we want here (but really a nitpick)

Reply:
fixed